great_expectations/rule_based_profiler/domain_builder/domain_builder.py

```diff
@@ -112,6 +112,12 @@ def get_batch_id(
     def batch_request(self) -> Optional[Union[BatchRequest, RuntimeBatchRequest, dict]]:
         return self._batch_request
 
+    @batch_request.setter
+    def batch_request(
+        self, batch_request: Union[BatchRequest, RuntimeBatchRequest, dict]
+    ) -> None:
+        self._batch_request = batch_request
```

Comment:
@cdkini I think that this will not work, because `batch_request` in builders must be serializable.  Having said that, we can easily specify structures containing non-serializable data as arguments to the profiler's `run()` method.  Thanks.

Reply:
Fixed!